src/vmprof_main.h

```diff
@@ -175,11 +179,43 @@ static PY_THREAD_STATE_T * _get_pystate_for_this_thread(void) {
 }
 #endif
 
+#ifdef VMPROF_LINUX
+static void broadcast_signal_for_threads(pid_t pid)
```

Comment:
I do not see opendir in the list of signal safe functions. Neither is readdir_r and atoi. I might be wrong, but we need to double check if they are signal safe! Maybe there is another API that enumerates the thread ids of the process without asking the disk!